standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/HiveMetaStore.java

```diff
@@ -9723,7 +9732,6 @@ public static void main(String[] args) throws Throwable {
       Lock startLock = new ReentrantLock();
       Condition startCondition = startLock.newCondition();
       AtomicBoolean startedServing = new AtomicBoolean();
-      startMetaStoreThreads(conf, startLock, startCondition, startedServing);
```

Comment:
No they aren't. In fact startMetaStore() doesn't know about these threads (without this change that is). The only condition is that these threads should be started before we call tServe.serve() as explained by the comment in startMetaStore().

We can not start these threads unless we know that the current HMS is a leader HMS. That happens after socket is bound. That's why I had to push startMetaStoreThreads() inside startMetaStore(). The way it's placed now, we won't spend time in starting these threads when the bind fails or creating TServer thread pool fails for any reason. So, we are actually saving on some resources in case of failures.